src/middleware/block_traffic.rs

```diff
@@ -47,6 +47,7 @@ impl Handler for BlockTraffic {
             .iter()
             .any(|value| self.blocked_values.iter().any(|v| v == value));
         if has_blocked_value {
+            super::log_request::add_custom_metadata(req, "cause", "blocked");
```

Comment:
I would rename the cause to `blacklisted header`, and add `header_name` and `header_value`.

Reply:
I've added the `header_name` to the message.  I didn't add the header value to the message, because capturing that would make the code more complicated at least for now all the headers we block on should already be captured in the logs.